pgserver/stmt.go

```diff
@@ -259,3 +262,27 @@ func RemoveComments(query string) string {
 
 	return buf.String()
 }
+
+var (
+	pgCatalogRegex     *regexp.Regexp
+	initPgCatalogRegex sync.Once
+)
+
+// get the regex to match any table in pg_catalog in the query.
+func getPgCatalogRegex() *regexp.Regexp {
+	initPgCatalogRegex.Do(func() {
+		var tableNames []string
+		for _, table := range catalog.GetInternalTables() {
+			if table.Schema != "__sys__" {
+				continue
+			}
+			tableNames = append(tableNames, table.Name)
+		}
+		pgCatalogRegex = regexp.MustCompile(`(?i)\s+(?:pg_catalog\.)?(` + strings.Join(tableNames, "|") + `)`)
```

Comment:
Maybe we need to prefix the pattern with `(FROM|JOIN|,)` to avoid undesired replacement.